lightning/src/ln/msgs.rs

```diff
@@ -305,6 +305,58 @@ impl NetAddress {
 	}
 }
 
+/// A "set" of addresses which enforces that there can be only up to one of each net address type.
+pub struct NetAddressSet {
+	v4: Option<NetAddress>,
+	v6: Option<NetAddress>,
+	onion2: Option<NetAddress>,
+	onion3: Option<NetAddress>,
+}
+impl NetAddressSet {
+	/// Creates a new, empty, NetAddressSet
+	pub fn new() -> Self {
+		NetAddressSet { v4: None, v6: None, onion2: None, onion3: None }
+	}
+
+	/// Sets the IPv4 socket address in this set, overwriting any previous IPv4 socket addresses
+	/// (if any).
+	pub fn set_v4(&mut self, addr: [u8; 4], port: u16) {
+		self.v4 = Some(NetAddress::IPv4 { addr, port });
+	}
+	/// Sets the IPv6 socket address in this set, overwriting any previous IPv4 socket addresses
```

Comment:
1ba81bc

Fix documentation there and under, "Sets the IPV6 socket address in this set, overwriting any previous IPv4 (?) socket addresses"